packages/hash/frontend/src/pages/[account-slug]/types/entity-type/array-data-type-menu.tsx

```diff
@@ -111,7 +137,13 @@ export const ArrayDataTypeMenu: FunctionComponent<ArrayDataTypeMenuProps> = ({
         typeId="array"
         prefix={prefix}
         deleteTooltip={deleteTooltip}
-        onDelete={onDelete}
+        onDelete={() => {
+          if (dataTypeCount + arrayCount + propertyObjectCount) {
```

Comment:
This is probably clearer
```suggestion
          if (dataTypeCount + arrayCount + propertyObjectCount > 0) {
```